tests/unit/scheduler.c

```diff
@@ -72,250 +56,101 @@ int verify_schedule(nccl_net_ofi_schedule_t *schedule, nccl_net_ofi_schedule_t *
+	* bytes are assigned 2 rail multiplexing */
+	schedule = scheduler->get_schedule(scheduler, min_stripe_size + 1, num_rails);
+	msg_size = NCCL_OFI_DIV_CEIL(NCCL_OFI_DIV_CEIL(min_stripe_size + 1, 2), align) * align;
+	if (!schedule) {
+		NCCL_OFI_WARN("Failed to get schedule");
 		free(ref_schedule);
-		return ret;
+		return -1;
 	}
 	ref_schedule->num_xfer_infos = 2;
-	ref_schedule->rail_xfer_infos[0].rail_id = 0;
+	ref_schedule->rail_xfer_infos[0].rail_id = 2;
 	ref_schedule->rail_xfer_infos[0].offset = 0;
-	ref_schedule->rail_xfer_infos[0].msg_size = 2 * align;
-	ref_schedule->rail_xfer_infos[1].rail_id = 1;
-	ref_schedule->rail_xfer_infos[1].offset = 2 * align;
-	ref_schedule->rail_xfer_infos[1].msg_size = 2 * align - 1;
-	ret = verify_schedule(schedule, ref_schedule);
-	if (ret) {
-		NCCL_OFI_WARN("Verification failed");
-		free(ref_schedule);
-		return ret;
-	}
-	free(schedule);
+	ref_schedule->rail_xfer_infos[0].msg_size = msg_size;
 
-	/* Data size = 4 * align */
-	num_rails = 3;
-	align = 3;
-	size = 4 * align;
-	ret = create_multiplexed(size, num_rails, align, &schedule);
-	if (ret) {
-		NCCL_OFI_WARN("Failed to create multiplexed schedule");
-		free(ref_schedule);
-		return ret;
-	}
-	ref_schedule->num_xfer_infos = 2;
-	ref_schedule->rail_xfer_infos[0].rail_id = 0;
-	ref_schedule->rail_xfer_infos[0].offset = 0;
-	ref_schedule->rail_xfer_infos[0].msg_size = 2 * align;
-	ref_schedule->rail_xfer_infos[1].rail_id = 1;
-	ref_schedule->rail_xfer_infos[1].offset = 2 * align;
-	ref_schedule->rail_xfer_infos[1].msg_size = 2 * align;
-	ret = verify_schedule(schedule, ref_schedule);
-	if (ret) {
-		NCCL_OFI_WARN("Verification failed");
-		free(ref_schedule);
-		return ret;
-	}
-	free(schedule);
+	ref_schedule->rail_xfer_infos[1].rail_id = 3;
+	ref_schedule->rail_xfer_infos[1].offset = msg_size;
+	ref_schedule->rail_xfer_infos[1].msg_size = min_stripe_size + 1 - msg_size;
 
-	/* Data size = 4 * align + 1 */
-	num_rails = 3;
-	align = 3;
-	size = 4 * align + 1;
-	ret = create_multiplexed(size, num_rails, align, &schedule);
-	if (ret) {
-		NCCL_OFI_WARN("Failed to create multiplexed schedule");
-		free(ref_schedule);
-		return ret;
-	}
-	ref_schedule->num_xfer_infos = 3;
-	ref_schedule->rail_xfer_infos[0].rail_id = 0;
-	ref_schedule->rail_xfer_infos[0].offset = 0;
-	ref_schedule->rail_xfer_infos[0].msg_size = 2 * align;
-	ref_schedule->rail_xfer_infos[1].rail_id = 1;
-	ref_schedule->rail_xfer_infos[1].offset = 2 * align;
-	ref_schedule->rail_xfer_infos[1].msg_size = 2 * align;
-	ref_schedule->rail_xfer_infos[2].rail_id = 2;
-	ref_schedule->rail_xfer_infos[2].offset = 4 * align;
-	ref_schedule->rail_xfer_infos[2].msg_size = 1;
 	ret = verify_schedule(schedule, ref_schedule);
 	if (ret) {
 		NCCL_OFI_WARN("Verification failed");
 		free(ref_schedule);
 		return ret;
 	}
-	free(schedule);
-
-	free(ref_schedule);
-
-	return 0;
-}
-
-int test_threshold_scheduler()
```

Comment:
Loop over the number of rails being tested with one block for the schedule verification? The four blocks are repetitive now.

Reply:
I refactored the tests so we check for 6 different msg sizes on each block. (where we initially tested one msg size  for number of rails)
We now test for below message sizes.
```
1. min_stripe_size + 1
2. min_stripe_size + align - 1
3. min_stripe_size + align
4. min_stripe_size + align + 1
5. 2*min_stripe_size - 1
6. 2*min_stripe_size
```
Although if we loop over the num_rails, the code would be more compact, it would make the tests a bit harder to understand I believe.